### PR TITLE
First steps for no static ip needed

### PR DESCRIPTION
**Pull request by Cabecinha84**
Changed flux coms.
This should be released when we release new bins.

### CODE REVIEW

**Comment by Cabecinha84** [comment]
If people don't have new bins installed, it will act like before.

**Comment by Cabecinha84** [comment]
So this can be reviewed and merged.

File: ZelBack/src/services/benchmarkService.js [hunk 1]
```diff
@@ -139,6 +139,14 @@ async function getInfo(req, res) {
   return res ? res.json(response) : response;
 }
 
+async function getBenchPublicIp(req, res) {
```
Comment (TheTrunk) [inline ZelBack/src/services/benchmarkService.js:142]: please rename to getPublicIp so it is the same name as the rpc method and different function name to the function in fluxCommunicaation

File: ZelBack/src/services/fluxCommunication.js [hunk 2]
```diff
@@ -93,6 +93,28 @@ async function getMyFluxIP() {
   return myIP;
 }
 
+async function getBenchPublicIp() {
```
Comment (TheTrunk) [inline ZelBack/src/services/fluxCommunication.js:96]: this function getBenchPublicIp and restartNodeBenchMarks are actually not needed, can be removed and simplified.
Please paste similar code directly to checkMyFluxAvailability where you are calling the 
await daemonService.restartNodeBenchmarks(); directly

File: ZelBack/src/services/fluxCommunication.js [hunk 3]
```diff
@@ -1081,7 +1103,14 @@ async function checkMyFluxAvailability(nodelist) {
     }
     if (resMyAvailability.data.status === 'error' || resMyAvailability.data.data.message.includes('not')) {
       log.error(`My Flux unavailability detected from ${askingIP}`);
-      // Asked Flux cannot reach me
+      // Asked Flux cannot reach me lets check if ip changed
+      const publicIp = await getBenchPublicIp();
+      if(publicIp && publicIp !== myIP){
+        myIP = publicIp;
+        await restartNodeBenchMarks();
+        await serviceHelper.delay(2*60*1000); //lets wait two minutes
```
Comment (TheTrunk) [inline ZelBack/src/services/fluxCommunication.js:1111]: spaces between * for readability

**Review by TheTrunk — CHANGES_REQUESTED** [review]
Concept ack, cosmetic and simplification changes requested. 
Shall be merged/released AFTER fork when we are sure such a method exists (bench 2.0.0 is mandatory)

File: ZelBack/src/services/benchmarkService.js (hunk 1 above)
Comment (Cabecinha84) [reply ZelBack/src/services/benchmarkService.js:142]: Adjusted.

File: ZelBack/src/services/fluxCommunication.js (hunk 3 above)
Comment (Cabecinha84) [reply ZelBack/src/services/fluxCommunication.js:1111]: Done

File: ZelBack/src/services/fluxCommunication.js (hunk 2 above)
Comment (Cabecinha84) [reply ZelBack/src/services/fluxCommunication.js:96]: Sorry, this part didn't understood

**Comment by Cabecinha84** [comment]
Tadeas I would like to have this on the version 1.14.0 that you have made the PR.
It's not mandatory for having new bench version installed. Has you can see if there is no success calling the getpublicip it will mark the DOS status like it was doing before these new calls.

**Review by TheTrunk — APPROVED** [review]

**Pull request merged**